src/main/java/jp/co/soramitsu/sora/sdk/crypto/json/SelectiveDisclosureFactory.java

```diff
@@ -46,28 +46,28 @@ public SelectiveDisclosureFactory() {
     this.merkleTreeFactory = new MerkleTreeFactory(digest);
   }
 
-
   /**
-   * Create cryptographic commitment for given input.
+   * Create cryptographic commitment for given raw input.
    *
    * @param input can be a Map<String, Object>, POJO, JsonNode; anything that {@link ObjectMapper}
-   * can process
+   *     can process
    * @return single selective disclosure item, which is used to create cryptographic proofs
    * @throws IOException is thrown when {@link ObjectMapper} can not process input JSON
    */
   public SelectiveDisclosureItem createCommitment(Object input) throws IOException {
-    JsonNode json = mapper.valueToTree(input);
-
-    if (!json.isObject()) {
-      throw new NotJsonObjectException(json);
-    }
-
-    ObjectNode obj = flattener.flatten((ObjectNode) json);
-    ObjectNode saltified = (ObjectNode) saltifier.saltify(obj);
-    List<Hash> hashes = hashifier.hashify(saltified);
-    MerkleTree tree = merkleTreeFactory.createFromLeaves(hashes);
+    return getSelectiveDisclosureItemFromSaltified(getSaltified(getNodeFromInput(input)));
+  }
 
-    return new SelectiveDisclosureItem(tree, saltified);
+  /**
+   * Create cryptographic commitment for given raw input.
+   *
+   * @param input can be a Map<String, Object>, POJO, JsonNode; anything that {@link ObjectMapper}
+   *     can process
+   * @return single selective disclosure item, which is used to create cryptographic proofs
+   * @throws IOException is thrown when {@link ObjectMapper} can not process input JSON
+   */
+  public SelectiveDisclosureItem createCommitmentFromSaltified(Object input) throws IOException {
```

Comment:
Better receive ObjectNode, because Object will most commonly be POJO.
And this method will be called on ObjectNode (or JsonNode)